community_images/common/orchestrator/k8s_setup.py

```diff
@@ -57,48 +57,58 @@ def __enter__(self):
         # bring up the k8s resources
         self.create_k8s_cluster()
 
-        # waiting for pod to be ready
-        logging.info("waiting for pod to be ready")
-        wait_complete = False
-        try:
-            readiness_wait_deployments_suffix = self.runtime_props.get(
-                "readiness_wait_deployments_suffix")
-            if not readiness_wait_deployments_suffix:
-                cmd = f"kubectl wait deployments {self.release_name}"
-                cmd += f" -n {self.namespace_name}"
-                cmd += " --for=condition=Available=True --timeout=20m"
-                Utils.run_cmd(cmd.split())
-            else:
-                for deployment_suffix in readiness_wait_deployments_suffix:
-                    cmd = f"kubectl wait deployments {self.release_name}-{deployment_suffix}"
-                    cmd += f" -n {self.namespace_name}"
-                    cmd += " --for=condition=Available=True --timeout=20m"
-                    Utils.run_cmd(cmd.split())
-
-            wait_complete = True
-        except subprocess.CalledProcessError:
-            logging.info(
-                "Wait deployment command failed, try waiting for pod-0")
+        # if health check script is available, use the health check script to wait
+        # for cluster health
+        if self.runtime_props.get("health_check_script", ""):
```

Comment:
should we call it readiness_check_script ? This will follow k8s guide as health check has different meaning.